### DISCUSSION TITLE
Syntax to allow assignment operator other than = on null

### DISCUSSION THREAD

**Discussion by Thaina** [dotnet/csharplang#314] [General]
Simply put, we might have nullable number. Operator assignment cannot be used there

```C#
int? x = null;
x += 1; // null here
```

I think we should allow `??` to work on the left side of `=`

```C#
int? x = null;
x ?? 0 += 1; // 1 here
```

The problem here is we might have long chain of member access to use operator assignment

Normally we need to do it like this

```C#
user.financeNullableDictionary["income"] = user.financeNullableDictionary["income"] ?? 0;
user.financeNullableDictionary["income"] += 10000; 
// or
user.financeNullableDictionary["income"] = (user.financeNullableDictionary["income"] ?? 0) + 10000; 
```

It should be shorter with `??`

```C#
user.financeNullableDictionary["income"] ?? 0 += 10000;
```

But it should not work in this case

```C#
user?.financeNullableDictionary?["income"] ?? 0 += 10000; // Should error?
```

**Comment by orthoxerox** [comment]
Why not

```c#
int? x = null;
x ??= 0;
x += 1;
```

If compound assignments become lvalues, you will be able to write `x ??= 0 += 1`

**Comment by Thaina** [comment]
I'm not sure but I already think that `x ??= 0 += 1` will became `x ??= 1` because rightmost = would be solved first
